admin-console/src/main/java/org/entando/entando/apsadmin/user/UserAction.java

```diff
@@ -32,297 +32,304 @@
+                    userProfile.setId(username);
+                    this.getUserProfileManager().addProfile(username, userProfile);
+                    return true;
+                }
+            } else {
+                return true;
+            }
+        } catch (Throwable t) {
+            logger.error("Error adding default profile for user {}", username, t);
+            throw new EntException("Error adding default profile for user " + username, t);
+        }
+        return false;
+    }
+
+    public String trash() {
+        try {
+            String result = this.checkUserForDelete();
+            if (null != result) {
+                return result;
+            }
+        } catch (Throwable t) {
+            logger.error("error in trash", t);
+            return FAILURE;
+        }
+        return SUCCESS;
+    }
+
+    public String delete() {
+        try {
+            String result = this.checkUserForDelete();
+            if (null != result) {
+                return result;
+            }
+            this.getUserManager().removeUser(this.getUsername());
+        } catch (Throwable t) {
+            logger.error("error in delete", t);
+            return FAILURE;
+        }
+        return SUCCESS;
+    }
+
+    protected boolean isCurrentUser() {
+        UserDetails currentUser = this.getCurrentUser();
+        return currentUser.getUsername().equals(this.getUsername());
+    }
+
+    protected boolean existsUser(String username) throws Throwable {
+        return (username != null) && (userManager.getUser(username) != null);
+    }
+
+    /**
+     * @deprecated Replaced by {@link #isEntandoUser(String)}
+     */
+    @Deprecated
+    protected boolean isJapsUser(String username) throws Throwable {
+        return this.isEntandoUser(username);
+    }
+
+    protected boolean isEntandoUser(String username) throws Throwable {
+        UserDetails user = this.getUserManager().getUser(username);
+        return (null != user && user.isEntandoUser());
+    }
+
+    protected String checkUserForEdit() throws Throwable {
+        if (!this.existsUser(this.getUsername())) {
+            this.addActionError(this.getText("error.user.notExist"));
+            return "userList";
+        }
+        if (!this.isEntandoUser(this.getUsername())) {
+            this.addActionError(this.getText("error.user.notLocal"));
+            return "userList";
+        }
+        return null;
+    }
+
+    protected String checkUserForDelete() throws Throwable {
+        if (!this.existsUser(this.getUsername()) && null == this.userProfileManager.getProfile(this.getUsername())) {
+            this.addActionError(this.getText("error.user.notExist"));
+            return "userList";
+        } else if (SystemConstants.ADMIN_USER_NAME.equals(this.getUsername())) {
+            this.addActionError(this.getText("error.user.cannotDeleteAdminUser"));
+            return "userList";
+        } else if (this.isCurrentUser()) {
```

Comment:
maybe here can be useful the same check existUser before to check if the user is the current

Reply:
if the username of the user to delete is that of the current user it is implied that it exists